pkg/util/hint/hint_processor.go

```diff
@@ -88,6 +88,18 @@ func ExtractTableHintsFromStmtNode(node ast.Node, sctx sessionctx.Context) []*as
 		// check duplicated hints
 		checkInsertStmtHintDuplicated(node, sctx)
 		return x.TableHints
+	case *ast.SetOprStmt:
+		var result []*ast.TableOptimizerHint
```

Comment:
Do we need to preallocate here? Might it not be worth it because usually we only have a few hints?

Reply:
It‘s hard to know how much memory do we need to allocate. 

BTW, Many statements are without hint so preallocate is unnecessary.